public/app/features/alerting/unified/utils/rulerClient.ts

```diff
@@ -157,16 +157,22 @@ export function getRulerClient(rulerConfig: RulerDataSourceConfig): RulerClient
       return addRuleToNamespaceAndGroup(namespace, groupSpec, newRule);
     }
 
-    const sameNamespace = existingRule.namespace === namespace;
-    const sameGroup = existingRule.group.name === values.group;
+    // we'll fetch the existing group again, someone might have updated it while we were editing a rule
+    const freshExisting = await findEditableRule(ruleId.fromRuleWithLocation(existingRule));
+    if (!freshExisting) {
+      throw new Error('Rule not found.');
```

Comment:
When is this possible??

Reply:
It shouldn't really – I tried to replicate this scenario but couldn't. Regardless it's typed as such and needs to be handled :D

Reply:
💯, I was only curious